### PR TITLE
ignore invalid If-Modified-Since and X-Qabel-Since headers

### PR DESCRIPTION
**Pull request by enkore**
Fixes https://github.com/Qabel/qabel-desktop/issues/501

### CODE REVIEW

File: tests/test_rest.py [hunk 1]
```diff
@@ -59,6 +59,14 @@ def test_get_messages_posted_since(self):
         assert response.status_code == status.HTTP_200_OK
         assert 'Hello World' in response.content.decode()
 
+    def test_get_messages_posted_since_invalid(self):
+        dt = datetime.datetime.now(tz=datetime.timezone.utc) - datetime.timedelta(minutes=1)
```
Comment (julianseeger) [inline tests/test_rest.py]: copy&paste waste

File: tests/test_rest.py [hunk 2]
```diff
@@ -59,6 +59,14 @@ def test_get_messages_posted_since(self):
...
+        response = self.app.get('/abcdefghijklmnopqrstuvwxyzabcdefghijklmnopo',
+                                HTTP_IF_MODIFIED_SINCE='warghabl')
+
+        assert response.status_code == status.HTTP_200_OK
+        assert 'Hello World' in response.content.decode()
```
Comment (julianseeger) [inline tests/test_rest.py]: why is no bar here but with an invalid HTTP_X_QABEL_NEW_SINCE there is a bar

Comment (enkore) [inline tests/test_rest.py]: it's always `bar not in`... "Bar" has a different Drop ID.

File: drop_service/views.py [hunk 3]
```diff
@@ -37,7 +40,12 @@ def _get_drops(self, drop_id):
         try:
             have_since, since = self.get_if_modified_since()
         except ValueError as value_error:
-            return error(str(value_error)), None
+            logger.warning('Could not parse modified-since header pack: %s', value_error)
+            have_since, since = False, None
+
+        if have_since and not since:
```
Comment (julianseeger) [inline drop_service/views.py]: on valueError, this wont be true because have_since is set to False. In which case is this logging?

Comment (enkore) [inline drop_service/views.py]: This block was on the wrong abstraction layer. I moved it to get_if_modified_since, which also makes it much more clear where this came from.

**Pull request merged**